trl/trainer/training_configs.py

```diff
@@ -39,3 +41,9 @@ class RewardConfig(TrainingArguments):
             "help": "The maximum length of the sequences in the batch. This argument is required if you want to use the default data collator."
         },
     )
+    gradient_checkpointing: Optional[bool] = field(
+        default=True,
```

Comment:
Here we override the default value of `transformers.TrainingArguments` to be `True` for backwards compatibility with how PEFT prepares k-bit models